src/lib.rs

```diff
@@ -47,21 +47,16 @@ impl StarReference {
         let (header, header_view) = generate_header(&settings.reference_path);
 
         // Load reference
-        let mut nvec = Vec::new();
-        for x in settings.args.iter() {
-            let cur_string = CString::new(x.as_str())?;
-            nvec.push(cur_string.into_raw());
-        }
-
-        let c_args = nvec.as_mut_ptr() as *mut *mut c_char;
+        let nvec = settings
+            .args
+            .iter()
+            .map(String::as_str)
+            .map(CString::new)
```

Comment:
```suggestion
            .map(|x| CString::new(x.as_str())
```
I'd be inclined to merge these two `map`.

Reply:
I find the lambda to be less clear than the chained map.  I'm curious which form generates more code, though - the form that requires construction of a lambda function, or the form that requires constructing an additional iterator object.